rio_tiler/io/cogeo.py

```diff
@@ -554,11 +566,18 @@ def feature(
         if not dst_crs:
             dst_crs = shape_crs
 
+        vrt_options = kwargs.pop("vrt_options", {})
+
         # Get BBOX of the polygon
         bbox = featureBounds(shape)
 
+        # If Dataset Origin is top Left (non-geo) we need to invert the bbox
+        # And set src_method="NO_GEOTRANSFORM" to use WarpedVRT
+        if self.bounds[1] > self.bounds[3] and not self.crs:
+            bbox = [bbox[0], bbox[3], bbox[2], bbox[1]]
+            vrt_options.update({"src_method": "NO_GEOTRANSFORM"})
```

Comment:
Kinda weird but we can use WarpedVRT with non geo dataset, we just need to add `src_method="NO_GEOTRANSFORM"` 

`Feature` is the only method that use WarpedVRT for read because of the cutline

Reply:
excellent, this is impossible in some R packages btw which is unhelpful since the warper is a supercharged RasterIO 👌